SimplenoteUITests/NoteList.swift

```diff
@@ -69,25 +69,9 @@ class NoteList {
 
     class func trashAllNotes() {
         NoteList.openAllNotes()
-
-        let notesNumber = NoteList.getNotesNumber()
-        let cellsNum = app.tables.element.children(matching: .cell).count
-        var startingIndex: Int
-
-        if notesNumber == cellsNum {
-            // Depending on what happened before, the cells numbering
-            // might not include "All Notes", "Trash" and "Settings" cells...
-            startingIndex = 0
-        } else {
-            // Or might include them
-            startingIndex = 3
-        }
-
-        for _ in 0..<notesNumber {
-            let cell = app.tables.cells.element(boundBy: startingIndex)
-            cell.swipeLeft()
-            cell.buttons[UID.Button.itemTrash].tap()
-        }
+        let noteNames = Table.getVisibleLabelledCellsNames()
+        guard noteNames.isEmpty == false else { return }
+        noteNames.forEach { Table.trashCell(noteName: $0) }
```

Comment:
If `noteNames` is empty, `forEach` will run on no notes. That's one of the many things I like about chaining operations on collections, you don't need to handle the case of an empty collection; if it's empty nothing will happen.

This simplifies (IMHO) the code:

```suggestion
        Table
            .getVisibleLabelledCellsNames()
            .forEach { Table.trashCell(noteName: $0) }
```

Reply:
This is extra nice, thank you!